sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/exemplar/FixedSizeExemplarReservoirTest.java

```diff
@@ -0,0 +1,135 @@
+    assertThat(reservoir.collectAndReset(Attributes.empty()))
+        .hasSize(1)
+        .satisfiesExactly(
+            exemplar ->
+                assertThat(exemplar)
+                    .hasEpochNanos(clock.now())
+                    .hasFilteredAttributes(Attributes.empty())
+                    .hasValue(1));
+
+    // Measurement count is reset, we should sample a new measurement (and only one)
+    clock.advance(Duration.ofSeconds(1));
+    reservoir.offerMeasurement(2L, Attributes.empty(), Context.root());
+    assertThat(reservoir.collectAndReset(Attributes.empty()))
+        .hasSize(1)
+        .satisfiesExactly(
+            exemplar ->
+                assertThat(exemplar)
+                    .hasEpochNanos(clock.now())
+                    .hasFilteredAttributes(Attributes.empty())
+                    .hasValue(2));
+  }
+
+  @Test
+  public void oneMeasurement_filtersAttributes() {
+    Attributes all =
+        Attributes.builder().put("one", 1).put("two", "two").put("three", true).build();
+    Attributes partial = Attributes.builder().put("three", true).build();
+    Attributes remaining = Attributes.builder().put("one", 1).put("two", "two").build();
+    TestClock clock = TestClock.create();
+    ExemplarReservoir reservoir =
+        new FixedSizeExemplarReservoir(clock, 1, RandomSupplier.platformDefault());
+    reservoir.offerMeasurement(1L, all, Context.root());
+    assertThat(reservoir.collectAndReset(partial))
+        .satisfiesExactly(
+            exemplar ->
+                assertThat(exemplar)
+                    .hasEpochNanos(clock.now())
+                    .hasValue(1)
+                    .hasFilteredAttributes(remaining));
+  }
+
+  @Test
+  public void oneMeasurement_includesTraceAndSpanIds() {
+    Attributes all =
+        Attributes.builder().put("one", 1).put("two", "two").put("three", true).build();
+    final Context context =
+        Context.root()
+            .with(
+                Span.wrap(
+                    SpanContext.createFromRemoteParent(
+                        TRACE_ID, SPAN_ID, TraceFlags.getSampled(), TraceState.getDefault())));
+    TestClock clock = TestClock.create();
+    ExemplarReservoir reservoir =
+        new FixedSizeExemplarReservoir(clock, 1, RandomSupplier.platformDefault());
+    reservoir.offerMeasurement(1L, all, context);
+    assertThat(reservoir.collectAndReset(Attributes.empty()))
+        .satisfiesExactly(
+            exemplar ->
+                assertThat(exemplar)
+                    .hasEpochNanos(clock.now())
+                    .hasValue(1)
+                    .hasFilteredAttributes(all)
+                    .hasTraceId(TRACE_ID)
+                    .hasSpanId(SPAN_ID));
+  }
+
+  @Test
+  public void multiMeasurements_preservesLatestSamples() {
+    AttributeKey<Long> key = AttributeKey.longKey("K");
+    // We cannot mock random in latest jdk, so we create an override.
```

Comment:
curious what changed with this. Any ideas?

Reply:
The error showed bytebuddy looking up some name that was considered JDK internal in 17.  I'm not really sure what it was, just decided to immediately workaround.


Here's the error:

```
org.mockito.exceptions.base.MockitoException: 
    Mockito cannot mock this class: class java.util.Random.

    Mockito can only mock non-private & non-final classes.
    If you're not sure why you're getting this error, please report to the mailing list.


    Java               : 17
    JVM vendor name    : Azul Systems, Inc.
    JVM vendor version : 17-ea+35
    JVM name           : OpenJDK 64-Bit Server VM
    JVM version        : 17-ea+35
    JVM info           : mixed mode, sharing
    OS name            : Mac OS X
    OS version         : 10.15.7


    Underlying exception : java.lang.IllegalStateException: Cannot access annotation property public abstract java.lang.String jdk.internal.util.random.RandomSupport$RandomGeneratorProperties.name()
        at app//io.opentelemetry.sdk.metrics.exemplar.FixedSizeExemplarReservoirTest.multiMeasurements_preservesLatestSamples(FixedSizeExemplarReservoirTest.java:113)

        Caused by:
        java.lang.IllegalStateException: Cannot access annotation property public abstract java.lang.String jdk.internal.util.random.RandomSupport$RandomGeneratorProperties.name()
```

Reply:
interesting. I wonder if it's a mockito bug that they know about.